model/project.go

```diff
@@ -1450,8 +1476,15 @@ func (p *Project) BuildProjectTVPairs(patchDoc *patch.Patch, alias string) {
 // mapping of the build variant to the tasks that will run on that build
 // variant. If includeDeps is set, it will also resolve task dependencies.
 func (p *Project) ResolvePatchVTs(patchDoc *patch.Patch, requester, alias string, includeDeps bool) (resolvedBVs []string, resolvedTasks []string, vts []patch.VariantTasks) {
-	bvs := patchDoc.BuildVariants
-	tasks := patchDoc.Tasks
+	var bvs, tasks []string
+	for _, bv := range patchDoc.BuildVariants {
+		bvRegex, _ := regexp.Compile(bv)
```

Comment:
Ignoring errors here seems like a code smell. If this is an invalid regex, shouldn't this return the error and stop?

Reply:
it looks like the function still returns whatever it has even with errors so i will grip the error and continue

Reply:
I won't say this is an invalid way to do it and I'm okay the way it is, but for the future, it's generally preferred to return an error to the layer where it's actually relevant rather than just log and move on. For example, if someone submitted a patch and wanted to know what went wrong, we wouldn't know to look for this error since it's not associated with any of the patch intent job logs.

At the minimum, can we include the patch doc ID in the log for debugging?

Reply:
added validation right before the function is called so in theory it should never hit it here but added patchdoc id anyway